homework_6/задания 2 уровня/3-рейтинг/rating.html

```diff
@@ -0,0 +1,85 @@
+    <title>Document</title>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css" integrity="sha512-1PKOgIY59xJ8Co8+NE6FZ+LOAZKjy+KY8iq0G4B3CyeY6wYHN3yt9PW0XpSriVlkMXe40PTKnXrLnZ9+fkDaog==" crossorigin="anonymous" />
+    <style>
+        i {
+            font-size: 50px;
+        }
+        
+        .golden {
+            color: gold;
+        }
+    </style>
+</head>
+
+<body>
+    <i id="first" class="fas fa-star"></i>
+    <i id="second" class="fas fa-star"></i>
+    <i id="third" class="fas fa-star"></i>
+    <i id="fourth" class="fas fa-star"></i>
+    <i id="fifth" class="fas fa-star"></i>
+
+
+    <!-- Задание. Используя заготовку, сделайте чтобы при наведении на вторую звезду
+    подсвечивались первая и вторая звезды. При наведени на третью звезду подсвечивались
+    первая, вторая и третья звезды, и т.д. 
+    mouseover - это событие срабатывает когда мы наводим на элемент.
+    mouseout - это событие срабатывает когда мы перестаем наводить на элемент.
+	Вы можете пойти более продвинутым путем, и получать предыдущий элемент 
+	например с помощью https://mzl.la/3a4SPlR
+    -->
+
+    <script>
+        // let firstStar = document.getElementById('first');
+        // let secondStar = document.getElementById('second');
+        // let thirdStar = document.getElementById('third');
+        // let fourthStar = document.getElementById('fourth');
+        // let fifthStar = document.getElementById('fifth');
+
+        let goldenClass = 'golden';
+
+        // console.log(fifthStar.previousElementSibling)
+
+        // firstStar.addEventListener('mouseover', function() {
+        //     if (!firstStar.classList.contains('goldenClass')) {
+        //         firstStar.classList.add(goldenClass);
+        //     }
+        // });
+
+        // firstStar.addEventListener('mouseout', function() {
+        //     firstStar.classList.remove(goldenClass);
+        // });
+
+        let iList = document.querySelectorAll("i");
+
+
+        window.addEventListener('mouseover', (event) => {
+            if ([].indexOf.call(iList, event.target) !== -1) {
+                event.target.classList.add(goldenClass)
+                if (event.target.id !== "first") {
+                    for (let i = [].indexOf.call(iList, event.target); i > 0; i--) {
+                        iList[i].previousElementSibling.classList.add(goldenClass);
+                    }
+                }
+            }
+        });
+
+        window.addEventListener('mouseout', (event) => {
+            if ([].indexOf.call(iList, event.target) !== -1) {
+                event.target.classList.remove(goldenClass)
+                if (event.target.id !== "first") {
+                    for (let i = [].indexOf.call(iList, event.target); i > 0; i--) {
+                        iList[i].previousElementSibling.classList.remove(goldenClass);
+                    }
+                }
+            }
```

Comment:
Здесь аналогичная ситуация, также подходит конструкция только с удалением класса
```suggestion
            if (event.target.classList.contains('fa-star')) {
                let currStar = event.target
                do {
                    currStar.classList.remove(goldenClass)
                }
                while (currStar = currStar.previousElementSibling)
            }
```